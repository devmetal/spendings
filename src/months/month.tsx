import React, { FC } from 'react';
import Card from 'react-bootstrap/Card';
import Button from 'react-bootstrap/Button';
import { SpendingsControl } from '../spendings/spendingsControl';
import { TSpending } from '../types';

export interface MonthProps {
  startedAt?: Date;
  endedAt?: Date;
  spendings: Array<TSpending>;
}

export const Month: FC<MonthProps> = ({ startedAt, endedAt, spendings }) => (
  <Card>
    <Card.Header>
      {(startedAt && (
        <React.Fragment>
          Started: {startedAt.toLocaleDateString()}
        </React.Fragment>
      )) || <React.Fragment>Start a new period</React.Fragment>}
      {endedAt && (
        <React.Fragment>Ended: {endedAt.toLocaleDateString()}</React.Fragment>
      )}
    </Card.Header>
    <SpendingsControl
      spendings={spendings}
      onCreate={() => {}}
      onDelete={() => {}}
      onUpdate={() => {}}
    />
    <Card.Body>
      {(endedAt && (
        <React.Fragment>
          Ended: {endedAt.toLocaleDateString()}
          <Button>See the results</Button>
        </React.Fragment>
      )) || (
        <React.Fragment>
          {(startedAt && <Button>End this period</Button>) || (
            <Button>Start this period</Button>
          )}
        </React.Fragment>
      )}
    </Card.Body>
  </Card>
);
