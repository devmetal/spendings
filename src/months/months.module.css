.Container {
  width: 100%;
  height: 100vh;
  overflow: hidden;
  position: relative;
  user-select: none;
}
