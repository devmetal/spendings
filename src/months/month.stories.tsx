import { Meta, Story } from '@storybook/react/types-6-0';
import { Month, MonthProps } from './month';

export default {
  title: 'money/Month',
  component: Month,
} as Meta;

const Template: Story<MonthProps> = (args) => <Month {...args} />;

export const BasicStarted = Template.bind({});
BasicStarted.args = {
  startedAt: new Date(),
  spendings: [
    { id: '1', name: 'Tesz', amount: 500 },
    { id: '2', name: 'Tesz', amount: 500 },
    { id: '3', name: 'Tesz', amount: 500 },
    { id: '4', name: 'Tesz', amount: 500 },
  ],
};

export const BasicEnded = Template.bind({});
BasicEnded.args = {
  startedAt: new Date(),
  endedAt: new Date(),
  spendings: [
    { id: '1', name: 'Tesz', amount: 500 },
    { id: '2', name: 'Tesz', amount: 500 },
    { id: '3', name: 'Tesz', amount: 500 },
    { id: '4', name: 'Tesz', amount: 500 },
  ],
};

export const Basic = Template.bind({});
Basic.args = {
  spendings: [],
};
