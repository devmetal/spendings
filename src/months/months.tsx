import React, { FC, useRef } from 'react';
import { useSprings, animated } from 'react-spring';
import { useGesture } from 'react-use-gesture';
import clamp from 'lodash/clamp';
import { Month, MonthProps } from './month';
import styles from './months.module.css';

export interface MonthsProps {
  months: Array<MonthProps>;
}

export const Months: FC<MonthsProps> = ({ months = [] }) => {
  const index = useRef(Math.max(months.length - 2, 0));

  const [props, set] = useSprings(months.length, (i) => ({
    x: (i - index.current) * window.innerWidth,
    display: i >= index.current - 1 && i <= index.current + 1,
  }));

  const bind = useGesture(
    {
      onDrag: ({
        down,
        movement: [xDelta],
        direction: [xDir],
        cancel = () => {},
      }) => {
        if (down && Math.abs(xDelta) > window.innerWidth * 0.25) {
          cancel();
          index.current = clamp(
            index.current + (xDir > 0 ? -1 : 1),
            0,
            months.length - 1
          );
        }

        set((i: number) => {
          if (i < index.current - 1 || i > index.current + 1) {
            return { display: false };
          }
          const x =
            (i - index.current) * window.innerWidth + (down ? xDelta : 0);
          return { x, display: true };
        });
      },
    },
    {
      drag: {
        lockDirection: true,
        filterTaps: true,
        axis: 'x',
      },
    }
  );

  return (
    <div className={styles.Container}>
      {props.map((item, index) => (
        <animated.div
          data-testid="month"
          key={index}
          style={{
            display: item.display.to((v) => (v ? 'block' : 'none')),
            transform: item.x.to((x) => `translate3d(${x}px,0,0)`),
          }}
          {...bind()}
        >
          <Month {...months[index]} />
        </animated.div>
      ))}
    </div>
  );
};
