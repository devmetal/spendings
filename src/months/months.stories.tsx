import { Meta, Story } from '@storybook/react/types-6-0';
import subDays from 'date-fns/subDays';
import { Months, MonthsProps } from './months';

export default {
  title: 'money/Months',
  component: Months,
} as Meta;

const Template: Story<MonthsProps> = (args) => <Months {...args} />;

export const Basic = Template.bind({});

const startDate = new Date();

Basic.args = {
  months: [
    {
      startedAt: subDays(startDate, 4),
      endedAt: subDays(startDate, 3),
      spendings: [
        { id: 7, name: 'Teszt', amount: 500 },
        { id: 8, name: 'Teszt', amount: 500 },
      ],
    },
    {
      startedAt: subDays(startDate, 3),
      endedAt: subDays(startDate, 2),
      spendings: [
        { id: 1, name: 'Teszt', amount: 500 },
        { id: 2, name: 'Teszt', amount: 500 },
      ],
    },
    {
      startedAt: subDays(startDate, 2),
      endedAt: subDays(startDate, 1),
      spendings: [
        { id: 3, name: 'Teszt', amount: 500 },
        { id: 4, name: 'Teszt', amount: 500 },
      ],
    },
    {
      spendings: [],
    },
  ],
};
