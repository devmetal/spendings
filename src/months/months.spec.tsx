import { render, screen } from '@testing-library/react';
import subDays from 'date-fns/subDays';
import { Months } from './months';

const startDate = new Date('2020.10.30');

const props = {
  months: [
    {
      startedAt: subDays(startDate, 4),
      endedAt: subDays(startDate, 3),
      spendings: [
        { id: '7', name: 'Teszt', amount: 500 },
        { id: '8', name: 'Teszt', amount: 500 },
      ],
    },
    {
      startedAt: subDays(startDate, 3),
      endedAt: subDays(startDate, 2),
      spendings: [
        { id: '1', name: 'Teszt', amount: 500 },
        { id: '2', name: 'Teszt', amount: 500 },
      ],
    },
    {
      startedAt: subDays(startDate, 2),
      endedAt: subDays(startDate, 1),
      spendings: [
        { id: '3', name: 'Teszt', amount: 500 },
        { id: '4', name: 'Teszt', amount: 500 },
      ],
    },
    {
      spendings: [],
    },
  ],
};

test('Sanity snapshot test', () => {
  const container = render(<Months {...props} />);
  expect(container).toMatchSnapshot();
});

test('Sanity snapshot test with 0 elements', () => {
  const container = render(<Months months={[]} />);
  expect(container).toMatchSnapshot();
});

describe('Starting element', () => {
  describe('when it has 1 month', () => {
    const testProps = {
      months: [props.months[3]],
    };

    it('the actual one avaibale month is visible', async () => {
      render(<Months {...testProps} />);
      const months = screen.getAllByTestId('month');
      expect(months[0].style.transform).toBe('translate3d(0px,0,0)');
    });
  });

  describe('when it has 2 month', () => {
    const testProps = {
      months: [props.months[2], props.months[3]],
    };

    it('the first element is visible', () => {
      render(<Months {...testProps} />);
      const months = screen.getAllByTestId('month');
      expect(months[0].style.transform).toBe('translate3d(0px,0,0)');
    });
  });

  describe('when it has 3 or more month', () => {
    it('the last active period visible', async () => {
      render(<Months {...props} />);
      const months = screen.getAllByTestId('month');
      expect(months[props.months.length - 2].style.transform).toBe(
        'translate3d(0px,0,0)'
      );
    });
  });
});
