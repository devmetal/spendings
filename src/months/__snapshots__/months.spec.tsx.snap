// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sanity snapshot test 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="Container"
      >
        <div
          data-testid="month"
          style="position: absolute; height: 100vh; width: 100%; display: none; transform: translate3d(-2048px,0,0);"
        >
          <div
            class="card"
          >
            <div
              class="card-header"
            >
              Started: 
              2020-10-26
              Ended: 
              2020-10-27
            </div>
            <div
              class="container"
            >
              <div
                class="row"
              >
                <div
                  class="container"
                >
                  <div
                    class="mb-3 input-group"
                  >
                    <div
                      class="input-group-prepend"
                    >
                      <span
                        class="input-group-text"
                      >
                        Name
                      </span>
                    </div>
                    <input
                      class="form-control"
                      data-testid="name"
                      type="text"
                      value=""
                    />
                    <div
                      class="input-group-prepend"
                    >
                      <span
                        class="input-group-text"
                      >
                        Amount
                      </span>
                    </div>
                    <input
                      class="form-control"
                      data-testid="amount"
                      min="1"
                      type="number"
                      value=""
                    />
                    <div
                      class="input-group-prepend"
                    >
                      <button
                        class="btn btn-outline-primary"
                        type="button"
                      >
                        Save
                      </button>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="row"
              >
                <div
                  class="container"
                >
                  <div
                    class="list-group"
                  >
                    <div
                      class="list-group-item"
                    >
                      <div
                        class="list-group list-group-horizontal"
                      >
                        <div
                          class="list-group-item"
                        >
                          Teszt
                        </div>
                        <div
                          class="list-group-item"
                        >
                          500
                        </div>
                        <div
                          class="list-group-item list-group-item-danger"
                        >
                          Delete
                        </div>
                      </div>
                    </div>
                    <div
                      class="list-group-item"
                    >
                      <div
                        class="list-group list-group-horizontal"
                      >
                        <div
                          class="list-group-item"
                        >
                          Teszt
                        </div>
                        <div
                          class="list-group-item"
                        >
                          500
                        </div>
                        <div
                          class="list-group-item list-group-item-danger"
                        >
                          Delete
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="card-body"
            >
              Ended: 
              2020-10-27
              <button
                class="btn btn-primary"
                type="button"
              >
                See the results
              </button>
            </div>
          </div>
        </div>
        <div
          data-testid="month"
          style="position: absolute; height: 100vh; width: 100%; display: block; transform: translate3d(-1024px,0,0);"
        >
          <div
            class="card"
          >
            <div
              class="card-header"
            >
              Started: 
              2020-10-27
              Ended: 
              2020-10-28
            </div>
            <div
              class="container"
            >
              <div
                class="row"
              >
                <div
                  class="container"
                >
                  <div
                    class="mb-3 input-group"
                  >
                    <div
                      class="input-group-prepend"
                    >
                      <span
                        class="input-group-text"
                      >
                        Name
                      </span>
                    </div>
                    <input
                      class="form-control"
                      data-testid="name"
                      type="text"
                      value=""
                    />
                    <div
                      class="input-group-prepend"
                    >
                      <span
                        class="input-group-text"
                      >
                        Amount
                      </span>
                    </div>
                    <input
                      class="form-control"
                      data-testid="amount"
                      min="1"
                      type="number"
                      value=""
                    />
                    <div
                      class="input-group-prepend"
                    >
                      <button
                        class="btn btn-outline-primary"
                        type="button"
                      >
                        Save
                      </button>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="row"
              >
                <div
                  class="container"
                >
                  <div
                    class="list-group"
                  >
                    <div
                      class="list-group-item"
                    >
                      <div
                        class="list-group list-group-horizontal"
                      >
                        <div
                          class="list-group-item"
                        >
                          Teszt
                        </div>
                        <div
                          class="list-group-item"
                        >
                          500
                        </div>
                        <div
                          class="list-group-item list-group-item-danger"
                        >
                          Delete
                        </div>
                      </div>
                    </div>
                    <div
                      class="list-group-item"
                    >
                      <div
                        class="list-group list-group-horizontal"
                      >
                        <div
                          class="list-group-item"
                        >
                          Teszt
                        </div>
                        <div
                          class="list-group-item"
                        >
                          500
                        </div>
                        <div
                          class="list-group-item list-group-item-danger"
                        >
                          Delete
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="card-body"
            >
              Ended: 
              2020-10-28
              <button
                class="btn btn-primary"
                type="button"
              >
                See the results
              </button>
            </div>
          </div>
        </div>
        <div
          data-testid="month"
          style="position: absolute; height: 100vh; width: 100%; display: block; transform: translate3d(0px,0,0);"
        >
          <div
            class="card"
          >
            <div
              class="card-header"
            >
              Started: 
              2020-10-28
              Ended: 
              2020-10-29
            </div>
            <div
              class="container"
            >
              <div
                class="row"
              >
                <div
                  class="container"
                >
                  <div
                    class="mb-3 input-group"
                  >
                    <div
                      class="input-group-prepend"
                    >
                      <span
                        class="input-group-text"
                      >
                        Name
                      </span>
                    </div>
                    <input
                      class="form-control"
                      data-testid="name"
                      type="text"
                      value=""
                    />
                    <div
                      class="input-group-prepend"
                    >
                      <span
                        class="input-group-text"
                      >
                        Amount
                      </span>
                    </div>
                    <input
                      class="form-control"
                      data-testid="amount"
                      min="1"
                      type="number"
                      value=""
                    />
                    <div
                      class="input-group-prepend"
                    >
                      <button
                        class="btn btn-outline-primary"
                        type="button"
                      >
                        Save
                      </button>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="row"
              >
                <div
                  class="container"
                >
                  <div
                    class="list-group"
                  >
                    <div
                      class="list-group-item"
                    >
                      <div
                        class="list-group list-group-horizontal"
                      >
                        <div
                          class="list-group-item"
                        >
                          Teszt
                        </div>
                        <div
                          class="list-group-item"
                        >
                          500
                        </div>
                        <div
                          class="list-group-item list-group-item-danger"
                        >
                          Delete
                        </div>
                      </div>
                    </div>
                    <div
                      class="list-group-item"
                    >
                      <div
                        class="list-group list-group-horizontal"
                      >
                        <div
                          class="list-group-item"
                        >
                          Teszt
                        </div>
                        <div
                          class="list-group-item"
                        >
                          500
                        </div>
                        <div
                          class="list-group-item list-group-item-danger"
                        >
                          Delete
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="card-body"
            >
              Ended: 
              2020-10-29
              <button
                class="btn btn-primary"
                type="button"
              >
                See the results
              </button>
            </div>
          </div>
        </div>
        <div
          data-testid="month"
          style="position: absolute; height: 100vh; width: 100%; display: block; transform: translate3d(1024px,0,0);"
        >
          <div
            class="card"
          >
            <div
              class="card-header"
            >
              Start a new period
            </div>
            <div
              class="container"
            >
              <div
                class="row"
              >
                <div
                  class="container"
                >
                  <div
                    class="mb-3 input-group"
                  >
                    <div
                      class="input-group-prepend"
                    >
                      <span
                        class="input-group-text"
                      >
                        Name
                      </span>
                    </div>
                    <input
                      class="form-control"
                      data-testid="name"
                      type="text"
                      value=""
                    />
                    <div
                      class="input-group-prepend"
                    >
                      <span
                        class="input-group-text"
                      >
                        Amount
                      </span>
                    </div>
                    <input
                      class="form-control"
                      data-testid="amount"
                      min="1"
                      type="number"
                      value=""
                    />
                    <div
                      class="input-group-prepend"
                    >
                      <button
                        class="btn btn-outline-primary"
                        type="button"
                      >
                        Save
                      </button>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="row"
              >
                <div
                  class="container"
                >
                  <div
                    class="list-group"
                  />
                </div>
              </div>
            </div>
            <div
              class="card-body"
            >
              <button
                class="btn btn-primary"
                type="button"
              >
                Start this period
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="Container"
    >
      <div
        data-testid="month"
        style="position: absolute; height: 100vh; width: 100%; display: none; transform: translate3d(-2048px,0,0);"
      >
        <div
          class="card"
        >
          <div
            class="card-header"
          >
            Started: 
            2020-10-26
            Ended: 
            2020-10-27
          </div>
          <div
            class="container"
          >
            <div
              class="row"
            >
              <div
                class="container"
              >
                <div
                  class="mb-3 input-group"
                >
                  <div
                    class="input-group-prepend"
                  >
                    <span
                      class="input-group-text"
                    >
                      Name
                    </span>
                  </div>
                  <input
                    class="form-control"
                    data-testid="name"
                    type="text"
                    value=""
                  />
                  <div
                    class="input-group-prepend"
                  >
                    <span
                      class="input-group-text"
                    >
                      Amount
                    </span>
                  </div>
                  <input
                    class="form-control"
                    data-testid="amount"
                    min="1"
                    type="number"
                    value=""
                  />
                  <div
                    class="input-group-prepend"
                  >
                    <button
                      class="btn btn-outline-primary"
                      type="button"
                    >
                      Save
                    </button>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="row"
            >
              <div
                class="container"
              >
                <div
                  class="list-group"
                >
                  <div
                    class="list-group-item"
                  >
                    <div
                      class="list-group list-group-horizontal"
                    >
                      <div
                        class="list-group-item"
                      >
                        Teszt
                      </div>
                      <div
                        class="list-group-item"
                      >
                        500
                      </div>
                      <div
                        class="list-group-item list-group-item-danger"
                      >
                        Delete
                      </div>
                    </div>
                  </div>
                  <div
                    class="list-group-item"
                  >
                    <div
                      class="list-group list-group-horizontal"
                    >
                      <div
                        class="list-group-item"
                      >
                        Teszt
                      </div>
                      <div
                        class="list-group-item"
                      >
                        500
                      </div>
                      <div
                        class="list-group-item list-group-item-danger"
                      >
                        Delete
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="card-body"
          >
            Ended: 
            2020-10-27
            <button
              class="btn btn-primary"
              type="button"
            >
              See the results
            </button>
          </div>
        </div>
      </div>
      <div
        data-testid="month"
        style="position: absolute; height: 100vh; width: 100%; display: block; transform: translate3d(-1024px,0,0);"
      >
        <div
          class="card"
        >
          <div
            class="card-header"
          >
            Started: 
            2020-10-27
            Ended: 
            2020-10-28
          </div>
          <div
            class="container"
          >
            <div
              class="row"
            >
              <div
                class="container"
              >
                <div
                  class="mb-3 input-group"
                >
                  <div
                    class="input-group-prepend"
                  >
                    <span
                      class="input-group-text"
                    >
                      Name
                    </span>
                  </div>
                  <input
                    class="form-control"
                    data-testid="name"
                    type="text"
                    value=""
                  />
                  <div
                    class="input-group-prepend"
                  >
                    <span
                      class="input-group-text"
                    >
                      Amount
                    </span>
                  </div>
                  <input
                    class="form-control"
                    data-testid="amount"
                    min="1"
                    type="number"
                    value=""
                  />
                  <div
                    class="input-group-prepend"
                  >
                    <button
                      class="btn btn-outline-primary"
                      type="button"
                    >
                      Save
                    </button>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="row"
            >
              <div
                class="container"
              >
                <div
                  class="list-group"
                >
                  <div
                    class="list-group-item"
                  >
                    <div
                      class="list-group list-group-horizontal"
                    >
                      <div
                        class="list-group-item"
                      >
                        Teszt
                      </div>
                      <div
                        class="list-group-item"
                      >
                        500
                      </div>
                      <div
                        class="list-group-item list-group-item-danger"
                      >
                        Delete
                      </div>
                    </div>
                  </div>
                  <div
                    class="list-group-item"
                  >
                    <div
                      class="list-group list-group-horizontal"
                    >
                      <div
                        class="list-group-item"
                      >
                        Teszt
                      </div>
                      <div
                        class="list-group-item"
                      >
                        500
                      </div>
                      <div
                        class="list-group-item list-group-item-danger"
                      >
                        Delete
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="card-body"
          >
            Ended: 
            2020-10-28
            <button
              class="btn btn-primary"
              type="button"
            >
              See the results
            </button>
          </div>
        </div>
      </div>
      <div
        data-testid="month"
        style="position: absolute; height: 100vh; width: 100%; display: block; transform: translate3d(0px,0,0);"
      >
        <div
          class="card"
        >
          <div
            class="card-header"
          >
            Started: 
            2020-10-28
            Ended: 
            2020-10-29
          </div>
          <div
            class="container"
          >
            <div
              class="row"
            >
              <div
                class="container"
              >
                <div
                  class="mb-3 input-group"
                >
                  <div
                    class="input-group-prepend"
                  >
                    <span
                      class="input-group-text"
                    >
                      Name
                    </span>
                  </div>
                  <input
                    class="form-control"
                    data-testid="name"
                    type="text"
                    value=""
                  />
                  <div
                    class="input-group-prepend"
                  >
                    <span
                      class="input-group-text"
                    >
                      Amount
                    </span>
                  </div>
                  <input
                    class="form-control"
                    data-testid="amount"
                    min="1"
                    type="number"
                    value=""
                  />
                  <div
                    class="input-group-prepend"
                  >
                    <button
                      class="btn btn-outline-primary"
                      type="button"
                    >
                      Save
                    </button>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="row"
            >
              <div
                class="container"
              >
                <div
                  class="list-group"
                >
                  <div
                    class="list-group-item"
                  >
                    <div
                      class="list-group list-group-horizontal"
                    >
                      <div
                        class="list-group-item"
                      >
                        Teszt
                      </div>
                      <div
                        class="list-group-item"
                      >
                        500
                      </div>
                      <div
                        class="list-group-item list-group-item-danger"
                      >
                        Delete
                      </div>
                    </div>
                  </div>
                  <div
                    class="list-group-item"
                  >
                    <div
                      class="list-group list-group-horizontal"
                    >
                      <div
                        class="list-group-item"
                      >
                        Teszt
                      </div>
                      <div
                        class="list-group-item"
                      >
                        500
                      </div>
                      <div
                        class="list-group-item list-group-item-danger"
                      >
                        Delete
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="card-body"
          >
            Ended: 
            2020-10-29
            <button
              class="btn btn-primary"
              type="button"
            >
              See the results
            </button>
          </div>
        </div>
      </div>
      <div
        data-testid="month"
        style="position: absolute; height: 100vh; width: 100%; display: block; transform: translate3d(1024px,0,0);"
      >
        <div
          class="card"
        >
          <div
            class="card-header"
          >
            Start a new period
          </div>
          <div
            class="container"
          >
            <div
              class="row"
            >
              <div
                class="container"
              >
                <div
                  class="mb-3 input-group"
                >
                  <div
                    class="input-group-prepend"
                  >
                    <span
                      class="input-group-text"
                    >
                      Name
                    </span>
                  </div>
                  <input
                    class="form-control"
                    data-testid="name"
                    type="text"
                    value=""
                  />
                  <div
                    class="input-group-prepend"
                  >
                    <span
                      class="input-group-text"
                    >
                      Amount
                    </span>
                  </div>
                  <input
                    class="form-control"
                    data-testid="amount"
                    min="1"
                    type="number"
                    value=""
                  />
                  <div
                    class="input-group-prepend"
                  >
                    <button
                      class="btn btn-outline-primary"
                      type="button"
                    >
                      Save
                    </button>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="row"
            >
              <div
                class="container"
              >
                <div
                  class="list-group"
                />
              </div>
            </div>
          </div>
          <div
            class="card-body"
          >
            <button
              class="btn btn-primary"
              type="button"
            >
              Start this period
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Sanity snapshot test with 0 elements 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="Container"
      />
    </div>
  </body>,
  "container": <div>
    <div
      class="Container"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
